src/realm/sync/instruction_applier.cpp

```diff
@@ -573,7 +586,7 @@ void InstructionApplier::operator()(const Instruction::ArrayInsert& instr)
                 [&](Mixed value) {
                     if (value.is_null()) {
                         if (col.is_nullable()) {
-                            list.set_null(index);
+                            list.insert_null(index);
```

Comment:
This is a bugfix that the new sync tests revealed, but this bug doesn't appear to exist in the sync v10 branch, it looks like it was introduced with dictionary [support](https://github.com/realm/realm-core/commit/15ca5c1468887067242a2463aa504a997d0c97c0) @simonask can you confirm this change won't negatively affect dictionaries?

Reply:
Sounds right, probably a merge bug :)